src/rgw/rgw_op.cc

```diff
@@ -3705,6 +3750,152 @@ void RGWPutACLs::execute()
   }
 }
 
+static void get_lc_oid(struct req_state *s, string& oid)
+{
+  string shard_id = s->bucket.name + ':' +s->bucket.bucket_id;
+  int max_objs = (s->cct->_conf->rgw_lc_max_objs > HASH_PRIME)?HASH_PRIME:s->cct->_conf->rgw_lc_max_objs;
+  int index = ceph_str_hash_linux(shard_id.c_str(), shard_id.size()) % HASH_PRIME % max_objs;
```

Comment:
The extra % HASH_PRIME is unnecessary, you already picked min(HASH_PRIME, lc_max_objs).

Question: if the lc_max_objs is changed, won't the index (and therefore the oid) change?  This could cause the wrong entries to be used for an object?  If so, this needs to be fixed, or at the very least a fat comment added to the config saying it cannot be changed.